iothub/device/src/HsmAuthentication/HttpClientHelper.cs

```diff
@@ -19,19 +19,15 @@ internal static class HttpClientHelper
             Justification = "The caller of this method is in charge of disposing the HTTP client that is returned.")]
         public static HttpClient GetHttpClient(Uri providerUri)
         {
-            HttpClient client;
-
             if (providerUri.Scheme.Equals(HttpScheme, StringComparison.OrdinalIgnoreCase)
-                | providerUri.Scheme.Equals(HttpsScheme, StringComparison.OrdinalIgnoreCase))
+                || providerUri.Scheme.Equals(HttpsScheme, StringComparison.OrdinalIgnoreCase))
```

Comment:
@timtay-microsoft here's an example of what I was talking about

Reply:
Yeah that definitely seems accidental